### PR TITLE
DOCS: DefaultCredentialsError could be raised if credentials not supplied

### PR DESCRIPTION
**Pull request by tswast**
Also, use Napoleon-style docs for the changed docs.

### CODE REVIEW

File: bigquery/google/cloud/bigquery/client.py [hunk 1]
```diff
@@ -93,25 +93,28 @@ def from_api_repr(cls, resource):
...
-
-    :type _http: :class:`~requests.Session`
-    :param _http: (Optional) HTTP object to make requests. Can be any object
-                  that defines ``request()`` with the same interface as
-                  :meth:`requests.Session.request`. If not passed, an
-                  ``_http`` object is created that is bound to the
-                  ``credentials`` for the current object.
-                  This parameter should be considered private, and could
-                  change in the future.
+    Keyword Args:
```
Comment (jonparrott) [inline bigquery/google/cloud/bigquery/client.py]: Just `Args:` is fine.

File: core/google/cloud/client.py [hunk 2]
```diff
@@ -94,20 +94,23 @@ class Client(_ClientFactoryMixin):
...
-
-    :type _http: :class:`~requests.Session`
-    :param _http: (Optional) HTTP object to make requests. Can be any object
-                  that defines ``request()`` with the same interface as
-                  :meth:`requests.Session.request`. If not passed, an
-                  ``_http`` object is created that is bound to the
-                  ``credentials`` for the current object.
-                  This parameter should be considered private, and could
-                  change in the future.
+    Keyword Args:
```
Comment (jonparrott) [inline core/google/cloud/client.py]: likewise here.

**Comment by dhermes** [comment]
@tswast We've generally followed a "don't document exceptions that are raised transitively" rule, but I'm happy to break it if you have a compelling case.

At the very least we could merge this PR with the Google-style docstring changes.

**Comment by jonparrott** [comment]
> but I'm happy to break it if you have a compelling case.

I'm fine with this as it's one of the primary failure cases here.

**Comment by tswast** [comment]
Understood. The reason I'm documenting this is that the colab team want to know what the exception type is raised so they can catch it and do their own auth flow on errors.

**Comment by jonparrott** [comment]
Will wait on CI to ensure docs still build

**Comment by jonparrott** [comment]
Build exceeded 10m, merging.

**Pull request merged**